src/main/java/com/develhope/spring/order/service/OrderService.java

```diff
@@ -12,9 +16,22 @@
 public class OrderService {
     @Autowired
     private OrderRepository orderRepository;
+    @Autowired
+    private VehicleRepository vehicleRepository;
 
-    public Order createOrder(Order order) {
-        return orderRepository.save(order);
+    public Order shopVehicle(Vehicle vehicle, VehicleState isPurchasable, User userId) {
+        if (vehicle == null || isPurchasable == null) {
```

Comment:
secondo me queste due exception non servono, sopratutto la orderNotFound.dovresti creartene una per questi casi , ad esempio InvalidOrder con i controlli .

Reply:
effettivamente è inutile un controllo su boolean, ma in ogni caso non serve creare l'eccezione in questione perchè questo metodo prenderà un id, che in caso non venga trovato, tirerà fuori l'eccezione vehiclenotfound o usernotfound